programs/mmm/src/ext_util.rs

```diff
@@ -0,0 +1,111 @@
+use anchor_lang::prelude::*;
+use solana_program::{account_info::AccountInfo, pubkey::Pubkey};
+use spl_token_2022::{
+    extension::{
+        group_member_pointer::GroupMemberPointer, metadata_pointer::MetadataPointer,
+        BaseStateWithExtensions, StateWithExtensions,
+    },
+    state::Mint as Token22Mint,
+};
+use spl_token_group_interface::state::TokenGroupMember;
+use spl_token_metadata_interface::state::TokenMetadata;
+
+use crate::{errors::MMMErrorCode, state::*};
+
+pub fn check_allowlists_for_mint_ext(
+    allowlists: &[Allowlist],
+    token_mint: &AccountInfo,
+    allowlist_aux: Option<String>,
```

Comment:
are we going to use merkle tree for t22? or token group is enough?

Reply:
should be good for now as no one is using the group extension